include/settings/settings.h

```diff
@@ -268,16 +268,18 @@ typedef int (*settings_load_direct_cb)(
  * It works as a blocking function, so it is up to the user to call
  * any kind of commit function when this operation ends.
  *
- * @param[in]     subtree subtree name of the subtree to be loaded.
- * @param[in]     cb      pointer to the callback function.
- * @param[in,out] param   parameter to be passed when callback
- *                        function is called.
+ * @param[in]     subtree           subtree name of the subtree to be loaded.
+ * @param[in]     cb                pointer to the callback function.
+ * @param[in,out] param             parameter to be passed when callback
+ *                                  function is called.
+ * @param[in]     filter_duplicates filter all duplicated entry.
```

Comment:
The proposed addition IMO does not justify an API change. Either enable it by default or use the filter_duplicate option only on settings_load_direct().

Reply:
It does not make a big sense to use it when all the settings are loaded at startup. It was the idea - do not change an external API. If somebody knows it is here and wishes to use it - set the specific field in the settings.

Reply:
You are changing an external API. Settings can be configured to use a custom backend, for these you are changing the API.

Reply:
@nvlsianpu - What is your opinion? Anybody more we can vote if it is worth to change API between settings and backend or is is better to add another backend function?
The first solution would fail during compilation if not adjusted. The second can lead to runtime error but if the functionality is not used, it would not require any adjustment from the user.

Reply:
@rakons, there are other options: enable it by default (makes the load a little longer) or use config to enable it.

Reply:
I would introduce this API change - it is no big change. Making it optional makes things harder to be maintained - I believe adaptation of a current custom back-end will be easy.

Reply:
@nvlsianpu, why not enable it by default ?

Reply:
This API was introduced like ~2 week ago by https://github.com/zephyrproject-rtos/zephyr/pull/19077

Reply:
@nvlsianpu ?? There is a new change used for passing the bool filter_duplicates.